massdriver-application-azure-app-service/main.tf

```
@@ -0,0 +1,138 @@
+module "application" {
+  source  = "github.com/massdriver-cloud/terraform-modules//massdriver-application?ref=a3371df"
+  name    = module.application.params.md_metadata.name_prefix
+  service = "function"
+}
+
+resource "azurerm_resource_group" "main" {
+  name     = var.md_metadata.name_prefix
+  location = var.vnet.specs.azure.region
+}
+
+resource "azurerm_service_plan" "main" {
+  name                = var.md_metadata.name_prefix
+  location            = azurerm_resource_group.main.location
+  resource_group_name = azurerm_resource_group.main.name
+
+  os_type      = "Linux"
+  sku_name     = var.application.sku_name
+  worker_count = var.application.minimum_worker_count
+}
+
+resource "azurerm_monitor_autoscale_setting" "main" {
+  name                = var.md_metadata.name_prefix
+  resource_group_name = azurerm_resource_group.main.name
+  location            = azurerm_resource_group.main.location
+  target_resource_id  = azurerm_service_plan.main.id
+  enabled             = true
+
+  profile {
+    name = "autoscale-profile"
+
+    capacity {
+      default = azurerm_service_plan.main.worker_count
+      minimum = azurerm_service_plan.main.worker_count
+      maximum = var.application.maximum_worker_count
+    }
+
+    rule {
+      metric_trigger {
+        metric_name        = "CpuPercentage"
+        metric_resource_id = azurerm_service_plan.main.id
+        time_grain         = "PT1M"
+        statistic          = "Average"
+        time_window        = "PT5M"
+        time_aggregation   = "Average"
+        operator           = "GreaterThan"
+        threshold          = 75
+      }
+      scale_action {
+        direction = "Increase"
+        type      = "ChangeCount"
+        value     = 1
+        cooldown  = "PT1M"
+      }
+    }
+
+    rule {
+      metric_trigger {
+        metric_name        = "CpuPercentage"
+        metric_resource_id = azurerm_service_plan.main.id
+        time_grain         = "PT5M"
+        statistic          = "Average"
+        time_window        = "PT10M"
+        time_aggregation   = "Average"
+        operator           = "LessThan"
+        threshold          = 25
+      }
+
+      scale_action {
+        direction = "Decrease"
+        type      = "ChangeCount"
+        value     = "1"
+        cooldown  = "PT1M"
+      }
+    }
+  }
+  notification {
+    email {
+      custom_emails = [var.application.notification_email]
+    }
+  }
+
+  depends_on = [
+    azurerm_service_plan.main
+  ]
+}
+
+resource "azurerm_linux_web_app" "main" {
+  name                = var.md_metadata.name_prefix
+  location            = azurerm_resource_group.main.location
+  resource_group_name = azurerm_resource_group.main.name
+  service_plan_id     = azurerm_service_plan.main.id
+
+  https_only = true
+
+  identity {
+    type = "SystemAssigned"
```

We might be able to use the service principal from the mdxc resource here. 

```suggestion
    type = "UserAssigned"
    identity_ids = [module.application.id]
```

---

`UserAssigned` identity is not the same as a service principal though. That's a whole 'nother thing, confusingly enough